frontend/src/components/DestructionListReviewer/DestructionListReviewer.tsx

```diff
@@ -31,51 +41,143 @@ export type DestructionListReviewerProps = {
 export function DestructionListReviewer({
   destructionList,
 }: DestructionListReviewerProps) {
-  const [modalState, setModalState] = useState<boolean>(false);
   const { state } = useNavigation();
   const revalidator = useRevalidator();
   const alert = useAlert();
+  const formDialog = useFormDialog();
   const reviewers = useReviewers();
+  const coReviewers = useCoReviewers();
+  const assignedCoReviewers = useDestructionListCoReviewers(destructionList);
   const user = useWhoAmI();
 
   /**
    * Gets called when the change is confirmed.
-   * @param _
-   * @param data
    */
-  const handleSubmit = (
-    _: FormEvent,
-    { reviewer, comment }: SerializedFormData,
-  ) => {
-    if (!modalState) {
-      return;
-    }
-    reassignDestructionList(destructionList.uuid, {
-      assignee: { user: Number(reviewer) },
-      comment: String(comment),
-    })
-      .then(revalidator.revalidate) // Reload the current route.
-      .catch(async (e) => {
+  const handleSubmit = (data: SerializedFormData) => {
+    const { coReviewer, reviewer, comment } = data as {
+      comment: string;
+      reviewer?: string;
+      coReviewer?: string[];
+    };
+
+    const promises: Promise<unknown>[] = [];
+
+    if (coReviewer) {
+      const add = coReviewer
+        .filter((pk) => Boolean(pk))
+        .map((pk) => ({ user: Number(pk) }));
+
+      const promise = updateCoReviewers(destructionList.uuid, {
+        add,
+        comment: comment as string,
+      }).catch(async (e) => {
         console.error(e);
+        try {
+          const data = await e.json();
+          const errors = collectErrors(data).join("\n");
+          alert("Foutmelding", data.detail || errors, "Ok");
+        } catch (e) {
+          alert(
+            "Foutmelding",
+            "Er is een fout opgetreden bij het bewerken van de mede beoordelaars!",
+            "Ok",
+          );
+          return;
+        }
+      });
 
-        const promise = e instanceof Response && e.json();
-        const data = await promise;
+      promises.push(promise);
+    }
 
-        alert(
-          "Foutmelding",
-          data.detail ||
+    const reviewerPk = Number(reviewer);
+    if (reviewerPk !== user?.pk) {
+      const promise = reassignDestructionList(destructionList.uuid, {
+        assignee: { user: reviewerPk },
+        comment: String(comment),
+      }).catch(async (e) => {
+        console.error(e);
+        try {
+          const data = await e.json();
+          const errors = collectErrors(data).join("\n");
+          alert("Foutmelding", data.detail || errors, "Ok");
+        } catch (e) {
+          alert(
+            "Foutmelding",
             "Er is een fout opgetreden bij het bewerken van de beoordelaar!",
-          "Ok",
-        );
+            "Ok",
+          );
+          return;
+        }
```

Comment:
These catch functions can be refactored for reuse? :thinking:

Reply:
- [ ] Reuse error handling in hooks.